org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/preferences/Preferences.java

```diff
@@ -203,6 +203,27 @@
 	public static final String JAVA_IMPORT_ORDER_KEY = "java.completion.importOrder";
 	public static final List<String> JAVA_IMPORT_ORDER_DEFAULT;
 
+	/**
+	 * A named preference that defines whether to use Objects.hash and
+	 * Objects.equals methods when generating the hashCode and equals methods.
+	 */
+	public static final String JAVA_HASHCODEEQUALSTEMPLATE_USEJAVA7OBJECTS = "java.hashCodeEqualsTemplate.useJava7Objects";
```

Comment:
I'd rather prefer you used java.codeGeneration.hashCodeEquals.*, as we'll most likely have other java.codeGeneration.* stuff in the future

Reply:
Accept the suggestion.